styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -328,34 +328,37 @@ private static boolean isActiveState(RunStateData runStateData) {
 
   private Response<Backfill> postBackfill(AuthContext ac, RequestContext rc,
       BackfillInput input) {
-    final BackfillBuilder builder = Backfill.newBuilder();
 
-    final String id = RandomGenerator.DEFAULT.generateUniqueId("backfill");
-
-    final WorkflowId workflowId = WorkflowId.create(input.component(), input.workflow());
-    final Workflow workflow;
     try {
-      workflow = storage.workflow(workflowId)
-          .orElseThrow(() -> new ResponseException(
-              Response.forStatus(Status.NOT_FOUND.withReasonPhrase("workflow not found"))));
+      return storage.runInTransactionWithRetries(tx -> postBackfill0(tx, ac, rc, input));
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
+  }
+
+  private Response<Backfill> postBackfill0(StorageTransaction tx, AuthContext ac, RequestContext rc,
+                                           BackfillInput input) throws IOException {
+    var builder = Backfill.newBuilder();
+
+    var id = RandomGenerator.DEFAULT.generateUniqueId("backfill");
+
+    var workflowId = WorkflowId.create(input.component(), input.workflow());
+    var workflow = tx.workflow(workflowId)
+        .orElseThrow(() -> new ResponseException(
+            Response.forStatus(Status.NOT_FOUND.withReasonPhrase("workflow not found"))));
 
     workflowActionAuthorizer.authorizeWorkflowAction(ac, workflow);
 
-    final Set<WorkflowInstance> activeWorkflowInstances;
-    try {
-      activeWorkflowInstances = storage.readActiveStates(input.component()).keySet();
-    } catch (Exception e) {
-      throw new RuntimeException(e);
-    }
+    var activeWorkflowInstances = storage.readActiveStates(input.component()).keySet();
 
-    final Optional<String> error = validate(rc, input, workflow);
-    if (error.isPresent()) {
-      return Response.forStatus(Status.BAD_REQUEST.withReasonPhrase(error.get()));
+    // Validate backfill & workflow
+    var validationError = validate(rc, input, workflow);
+    if (validationError.isPresent()) {
+      return Response.forStatus(Status.BAD_REQUEST.withReasonPhrase(validationError.get()));
     }
 
+    // Forbid backfill creation if there are already active instances in the backfill range
+    // TODO: Does this make sense?
```

Comment:
To me it makes sense, otherwise users can't know which run will be the final one and might get unexpected results. Could we elaborate on why it wouldn't make sense?

Reply:
We are already doing this: https://github.com/spotify/styx/pull/745/files#diff-0cec30db46b18b2c0132f746e3bad5b0R371

Or this refers to something else?

Reply:
NVM, this is a question, not a real todo. And yes, to me it makes sense as well.